### PR TITLE
Dev Tool: Plugin Test Server for Sigma

### PR DESCRIPTION
**Pull request by stephenrlouie**
Lets plugin devs more easily test their plugins against the sigma API, it also gives devs control over what goes into the inputs.

A plugin test server that knows how to serve files from a directory as `/input` and `/previous-input` it will also take every POST to `/output` and `/heartbeat` and write that body to a file in its `baseDir`

It also has support for Bearer tokens.

### CODE REVIEW

File: plugin/test-server/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Plugin Test Server
+
+This is a standalone HTTP server that serves files from a given directory to enable plugin developers to test plugins.
+The `/input` endpoint expects a file named `input` and the `/previous-input` expects a file named `previous-input`.
```
Comment (helik) [inline plugin/test-server/README.md]: "the `/previous-input` endpoint" or just `/previous-input`

File: plugin/test-server/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+inputs to the plugin. When a plugin POSTS an output it will be written to `./files/output` and `./files/output-<timestamp>`
+
+```docker run --name <HOSTNAME> -p 5643:5643 -v `pwd`/files:/exampleDir dockerhub.cisco.com/project-mu-release-docker/plugin-server:latest -dir /exampleDir```
+
+
+## Running your plugin (raw & containerized)
+Important Environment Variables.
+- `SIGMA_URL`: indicates where the test-server is running for example `127.0.0.1:5643`.
+- `SIGMA_HEARTBEAT_TTL`: tells the plugin what the timeout time sigma expects. We recommend
+heartbeating at `HEARTBEAT/2` or some smaller number, since if Sigma will consider your plugin dead if you checkin every `HEARTBEAT+1`
```
Comment (meganokeefe) [inline plugin/test-server/README.md]: checkin -> check in

File: plugin/test-server/client/.gitignore [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
```
Comment (meganokeefe) [inline plugin/test-server/client/.gitignore]: ?

File: plugin/test-server/client/docs/ServiceOperationOutput.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# ServiceOperationOutput
```
Comment (meganokeefe) [inline plugin/test-server/client/docs/ServiceOperationOutput.md]: why is this here but no other docs?

Comment (meganokeefe) [reply plugin/test-server/client/docs/ServiceOperationOutput.md]: https://github.com/sigma-dev/sigma/tree/testModeEngaged/plugin/test-server/client/docs

**Comment by meganokeefe** [comment]
We should add a doc for this to the sigma docs site.

**Comment by meganokeefe** [comment]
other than those comments, lgtm LGTM LGtM lGtM
